core/services/p2p/signer.go

```diff
@@ -0,0 +1,43 @@
+package p2p
+
+import (
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+type signer struct {
+	keystoreP2P keystore.P2P
+	peerID      p2pkey.PeerID
+	privateKey  crypto.Signer
+}
+
+var _ types.Signer = &signer{}
+
+func NewSigner(keystoreP2P keystore.P2P, peerID p2pkey.PeerID) *signer {
```

Comment:
@bolekk Why are we passing the keystore + peerID into this object, as opposed to fetching the key and passing that in. What we're doing here seems like we're coupling things together -- and the Initialize method being added also seems odd.

Reply:
Added a comment.